test/sanity/code-smell/mypy.json

```diff
@@ -0,0 +1,4 @@
+{
+    "no_targets": true,
```

Comment:
We shouldn't use `no_targets` for this test, since that will cause the test to always run.

We'll need to see how mypy behaves if given individual files instead of entire modules. If the results for a given file vary based on whether the filename is given vs its module, then we'll need to come up with something different than is normally used for sanity tests.

Reply:
@mattclay it is known that mypy needs to be pointed to whole packages (folders with modules, ideally, top-level). Otherwise, if you point to two modules with the same name in different folders, it'll error out with "duplicate module, go away".

Reply:
Also, mypy checks types in the context of other modules because they call functions from each other, and only by analysing everything the linter can say if everything is alright.